components/TokenManagePage/Media.tsx

```diff
@@ -51,6 +87,19 @@ const Media = () => {
               placeholder="enter a description"
             />
           </div>
+
+          <button
+            className="bg-black text-grey-eggshell w-fit px-8 py-2 rounded-md disabled:opacity-50"
+            onClick={onSave}
+            disabled={isLoading || isSaving || !isOwner}
+          >
+            {isSaving ? "saving..." : "Save"}
+          </button>
+          {!isOwner && (
+            <p className="text-xs text-grey-moss-500">
+              Only the contract owner can save changes.
+            </p>
+          )}
```

Comment:
Single Responsibility Principle
- actual: Media component defines HTML and logic for OwnerWarning
- required: new, standalone, component file for `<OwnerWarning />`